docker/docker-entrypoint.sh

```diff
@@ -3,37 +3,41 @@
 set -e
 
 # Get MuJoCo key from the environment
-echo "${MJKEY}" > /root/.mujoco/mjkey.txt
+echo -e "${MJKEY}" > /root/.mujoco/mjkey.txt
 
-# Setup dummy X server display
-display_num=0
-export DISPLAY=:"${display_num}"
-Xvfb "${DISPLAY}" -screen 0 1024x768x24 &
-pulseaudio -D --exit-idle-time=-1
+if [[ "${IMAGE_TYPE}" == "headless" ]]; then
+  # Setup dummy X server display
+  # Socket for display :0 may already be in use if the container is connected
+  # to the network of the host, that's why we use 1.
+  display_num=1
+  export DISPLAY=:"${display_num}"
+  Xvfb "${DISPLAY}" -screen 0 1024x768x24 &
+  pulseaudio -D --exit-idle-time=-1
 
-# Wait for X to come up
-file="/tmp/.X11-unix/X${display_num}"
-for i in $(seq 1 10); do
+  # Wait for X to come up
+  file="/tmp/.X11-unix/X${display_num}"
+  for i in $(seq 1 10); do
     if [ -e "$file" ]; then
       break
     fi
     echo "Waiting for X to start (i.e. $file to be created) (attempt $i/10)"
     sleep "$i"
-done
-if ! [ -e "$file" ]; then
+  done
+  if ! [ -e "$file" ]; then
     echo "Timed out waiting for X to start: $file was not created"
     exit 1
+  fi
+  export DISABLE_MUJOCO_RENDERING=1  # silence glfw
 fi
 
 # Activate conda environment
 source activate garage
 
 # Fixes Segmentation Fault
 # See: https://github.com/openai/mujoco-py/pull/145#issuecomment-356938564
-export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libGLEW.so:/usr/lib/x86_64-linux-gnu/mesa/libGL.so.1
+export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libGLEW.so
 
 export TF_CPP_MIN_LOG_LEVEL=3      # shut TensorFlow up
-export DISABLE_MUJOCO_RENDERING=1  # silence glfw
 export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${HOME}/.mujoco/mjpro150/bin"
 
-exec "$@"
+source /root/.bashrc && exec "$@"
```

Comment:
do not source a bashrc in a docker environment.

Reply:
I'm sourcing bashrc because python packages used in the ROS workspace and the intera_interface are loaded in that way. Otherwise, if we try to run a python script that imports `intera_interface` and/or `rospy`, it will not find them.
A way to avoid this could be an environment variable with a list of all the setup scripts, and then source each of the scripts in the entrypoint, but that list has to be carried on from one docker build to another in dockercompose.
Any ideas to solve this problem are really appreciated :smiley_cat:

Reply:
that problem should be dealt with in garage-ros, not here. it's bad form to be sourcing bashrc inside a Docker container.

I realize ROS is full of bad design decisions, but we shouldn't let them percolate into these unrelated containers.

for instance, your ROS container can just be FROM this image, but with a different entrypoint.